polar-core/src/debugger.rs

```diff
@@ -251,60 +249,8 @@ impl Debugger {
                 }
             }
             "stack" | "trace" => {
-                let mut trace_stack = vm.trace_stack.clone();
-                let mut trace = vm.trace.clone();
-
-                // Walk up the trace stack so we can print out the current query at each level.
-                let mut stack = vec![];
-                while let Some(t) = trace.last() {
-                    stack.push(t.clone());
-                    trace = trace_stack
-                        .pop()
-                        .map(|ts| ts.as_ref().clone())
-                        .unwrap_or_else(Vec::new);
-                }
-
-                stack.reverse();
-
-                // Only index queries, not rules. Rule nodes are just used as context for where the query comes from.
-                let mut i = stack.iter().filter(|t| t.term().is_some()).count();
-
-                let mut st = String::new();
-                let mut rule = None;
-                for t in stack {
-                    match &t.node {
-                        Node::Rule(r) => {
-                            rule = Some(r.clone());
-                        }
-                        Node::Term(t) => {
-                            if matches!(t.value(), Value::Expression(Operation { operator: Operator::And, args}) if args.len() == 1)
-                            {
-                                continue;
-                            }
-
-
-                            let _ = write!(st, "{}: {}", i-1, vm.term_source(t, false));
-                            i -= 1;
-                            let _ = write!(st, "\n  ");
-                            if let Some(source) = vm.source(t) {
-                                if let Some(rule) = &rule {
-                                    let _ = write!(st, "in rule {} ", rule.name);
-                                } else {
-                                    let _ = write!(st, "in query ");
-                                }
-                                let (row, column) = crate::lexer::loc_to_pos(&source.src, t.offset());
-                                let _ = write!(st, "at line {}, column {}", row + 1, column + 1);
-                                if let Some(filename) = source.filename {
-                                    let _ = write!(st, " in file {}", filename);
-                                }
-                                let _ = writeln!(st);
-                            };
-                        }
-                    }
-                }
-
                 return Some(Goal::Debug {
-                    message: st
+                    message: vm.stack_trace()
```

Comment:
Added test for previous `"stack"` debugger command output: 2e32bea8f694a138eac0db08b304cb511d1f876e

Reused stack trace formatting from `VM::stack_trace` for the debugger's `"stack"` command: 5c59c3474fc30a5c264240ebd290a290b0fbeb69

Reply:
Had to update a few tests: c332128f9